source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,289 @@
+        {
+            gm->getTile(ii,jj)->hide();
+        }
+    }
+}
+
+
+
+void CullingManager::showAllTiles(void)
+{
+    GameMap* gm = mCm->mGameMap;
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->show();
+        }
+    }
+}
+
+
+
+
+
+void CullingManager::newBashAndSplashTiles(int64_t mode){
+    int64_t xxLeftOld = oldWalk.getTopLeftVertex().x;
+    int64_t xxRightOld= oldWalk.getTopRightVertex().x;
+    int64_t xxLeft = mWalk.getTopLeftVertex().x;
+    int64_t xxRight= mWalk.getTopRightVertex().x;
+    int64_t DxRight, DxLeft;
+    int64_t xxp, yyp;
+    std::stringstream ss;
+    int64_t bb = (( std::min(mWalk.getBottomLeftVertex().y , oldWalk.getBottomRightVertex().y) >> mPrecisionDigits) - 2) << mPrecisionDigits;
+
+    for (int64_t yy = ((std::max(mWalk.getTopLeftVertex().y , oldWalk.getTopRightVertex().y  ) >> mPrecisionDigits) + 2) << mPrecisionDigits;  yy >= bb; yy -= Unit)
+    {
+        oldWalk.notifyOnMoveDown(yy);
+        mWalk.notifyOnMoveDown(yy);
+        DxLeft = mWalk.getCurrentXLeft(yy);
+        xxLeft = DxLeft;
+        xxLeftOld = oldWalk.getCurrentXLeft(yy);
+        DxRight = mWalk.getCurrentXRight(yy);
+        xxRight = DxRight;
+        xxRightOld = oldWalk.getCurrentXRight(yy);
+   
+        if(mDebug)
+        {
+            ss << endl;
+            ss << "DxLeft, DxRight: " << double(DxLeft)/Unit << " " << double(DxRight)/Unit << " " ;
+            LogManager::getSingleton().logMessage( ss.str(),LogMessageLevel::NORMAL);
+            ss.str("");
+            ss << "xxLeft " << (xxLeft  >> mPrecisionDigits) << "xxRight " << (xxRight  >> mPrecisionDigits );
+            LogManager::getSingleton().logMessage( ss.str(),LogMessageLevel::NORMAL);
+            ss.str("");
+        }
+    
+        int64_t mm = ((std::min(xxLeft, xxLeftOld) >> mPrecisionDigits) << mPrecisionDigits) ;
+        if(std::min(xxLeft, xxLeftOld) < max(xxRight,xxRightOld) )
+            for (int64_t xx = mm ; xx <= max(xxRight,xxRightOld) ; xx+= Unit)
+            {
+                bool bash = (xx >= xxLeftOld && xx <= xxRightOld && (yy >= oldWalk.getBottomLeftVertex().y) && yy <= oldWalk.getTopLeftVertex().y);
+                bool splash = (xx >= xxLeft && xx <= xxRight && (yy >= mWalk.getBottomLeftVertex().y) && yy <= mWalk.getTopLeftVertex().y);
+
+                if (mDebug)
+                {
+                    ss<< " x " <<  (xx >> mPrecisionDigits )<< " y " << (yy >> mPrecisionDigits ) << " " << (splash && (mode & SHOW)) << (bash && (mode & HIDE)) << endl;
+            
+                }
+                xxp = ((xx >>mPrecisionDigits) - 128.0);
+                yyp = ((yy >>mPrecisionDigits) - 128.0);
+                GameMap* gm = mCm->mGameMap;
+                if(bash && splash && (mode & HIDE) && (mode & SHOW))
+                {
+                    // Nothing
```

Comment:
True. Nevermind about this then. But if you add brackets to the first if, then please brackets to the others for better readability.